src/term/mod.rs

```diff
@@ -580,15 +581,16 @@ impl Term {
 
         // Clear `lines` lines starting from origin to origin + lines
         {
-            let start = origin;
-            let end = start + lines;
+            let start = limit(origin, Line(0), self.grid.num_lines() - 1);
+            let end = limit(start + lines, Line(0), self.grid.num_lines() - 1);
             self.grid.clear_region(start..end, |c| c.reset(&template));
         }
 
         // Scroll from origin to bottom less number of lines
         {
-            let start = origin;
-            let end = self.scroll_region.end - lines;
+            let start = limit(origin, Line(0), self.grid.num_lines() - 1);
+            let end = limit(self.scroll_region.end - lines, Line(0), self.grid.num_lines() - 1);
```

Comment:
I think that some of these can be avoided by ensuring things like the `scroll_region` are valid when being set and after resizing.